File: mihabitat-web/src/main/webapp/recursos/js/app/reportes/adeudo-meses-proveedor.js
var AdeudoMeses = function(data) {
	AdeudoProveedor.call(this, data);
    
    var self = this;
    
    self.enero = ko.observable();
    self.febrero = ko.observable();
    self.marzo = ko.observable();
    self.abril = ko.observable();
    self.mayo = ko.observable();
    self.junio = ko.observable();
    self.julio = ko.observable();
    self.agosto = ko.observable();
    self.septiembre = ko.observable();
    self.octubre = ko.observable();
    self.noviembre = ko.observable();
    self.diciembre = ko.observable();
    self.total = ko.observable();
    
    self.limpiarAdeudoMeses= function() {
        self.limpiar();
        
        self.enero(undefined);
        self.febrero(undefined);
        self.marzo(undefined);
        self.abril(undefined);
        self.mayo(undefined);
        self.junio(undefined);
        self.julio(undefined);
        self.agosto(undefined);
        self.septiembre(undefined);
        self.octubre(undefined);
        self.noviembre(undefined);
        self.diciembre(undefined);
        self.total(undefined);
    }
    
    self.cargarAdeudoMeses = function(data) {
        self.cargar(data);
        
        self.enero(data ? data.enero : undefined);
        self.febrero(data ? data.febrero : undefined);
        self.marzo(data ? data.marzo : undefined);
        self.abril(data ? data.abril : undefined);
        self.mayo(data ? data.mayo : undefined);
        self.junio(data ? data.junio : undefined);
        self.julio(data ? data.julio : undefined);
        self.agosto(data ? data.agosto : undefined);
        self.septiembre(data ? data.septiembre : undefined);
        self.octubre(data ? data.octubre : undefined);
        self.noviembre(data ? data.noviembre : undefined);
        self.diciembre(data ? data.diciembre : undefined);
        self.total(data ? data.total : undefined);
    }   
}

File: mihabitat-web/src/main/java/com/bstmexico/mihabitat/web/dto/reportes/AdeudoMesesProveedor.java
package com.bstmexico.mihabitat.web.dto.reportes;

import java.math.BigDecimal;

/**
 * @author Pablo Cruz Santos
 * @version 1.0
 * @since 2015
 */
public class AdeudoMesesProveedor extends AdeudoProveedor {

	private BigDecimal enero;

	private BigDecimal febrero;

	private BigDecimal marzo;

	private BigDecimal abril;

	private BigDecimal mayo;

	private BigDecimal junio;

	private BigDecimal julio;

	private BigDecimal agosto;

	private BigDecimal septiembre;

	private BigDecimal octubre;

	private BigDecimal noviembre;

	private BigDecimal diciembre;

	private BigDecimal total;

	public AdeudoMesesProveedor() {
		super();
		this.enero = BigDecimal.ZERO;
		this.febrero = BigDecimal.ZERO;
		this.marzo = BigDecimal.ZERO;
		this.abril = BigDecimal.ZERO;
		this.mayo = BigDecimal.ZERO;
		this.junio = BigDecimal.ZERO;
		this.julio = BigDecimal.ZERO;
		this.agosto = BigDecimal.ZERO;
		this.septiembre = BigDecimal.ZERO;
		this.octubre = BigDecimal.ZERO;
		this.noviembre = BigDecimal.ZERO;
		this.diciembre = BigDecimal.ZERO;
	}

	public BigDecimal getEnero() {
		return enero;
	}

	public void setEnero(BigDecimal enero) {
		this.enero = enero;
	}

	public BigDecimal getFebrero() {
		return febrero;
	}

	public void setFebrero(BigDecimal febrero) {
		this.febrero = febrero;
	}

	public BigDecimal getMarzo() {
		return marzo;
	}

	public void setMarzo(BigDecimal marzo) {
		this.marzo = marzo;
	}

	public BigDecimal getAbril() {
		return abril;
	}

	public void setAbril(BigDecimal abril) {
		this.abril = abril;
	}

	public BigDecimal getMayo() {
		return mayo;
	}

	public void setMayo(BigDecimal mayo) {
		this.mayo = mayo;
	}

	public BigDecimal getJunio() {
		return junio;
	}

	public void setJunio(BigDecimal junio) {
		this.junio = junio;
	}

	public BigDecimal getJulio() {
		return julio;
	}

	public void setJulio(BigDecimal julio) {
		this.julio = julio;
	}

	public BigDecimal getAgosto() {
		return agosto;
	}

	public void setAgosto(BigDecimal agosto) {
		this.agosto = agosto;
	}

	public BigDecimal getSeptiembre() {
		return septiembre;
	}

	public void setSeptiembre(BigDecimal septiembre) {
		this.septiembre = septiembre;
	}

	public BigDecimal getOctubre() {
		return octubre;
	}

	public void setOctubre(BigDecimal octubre) {
		this.octubre = octubre;
	}

	public BigDecimal getNoviembre() {
		return noviembre;
	}

	public void setNoviembre(BigDecimal noviembre) {
		this.noviembre = noviembre;
	}

	public BigDecimal getDiciembre() {
		return diciembre;
	}

	public void setDiciembre(BigDecimal diciembre) {
		this.diciembre = diciembre;
	}

	public BigDecimal getTotal() {
		total = getEnero().add(getFebrero()).add(getMarzo()).add(getAbril())
				.add(getMayo()).add(getJunio()).add(getJulio())
				.add(getAgosto()).add(getSeptiembre()).add(getOctubre())
				.add(getNoviembre()).add(getDiciembre());
		return total;
	}

	public void setTotal(BigDecimal total) {
		this.total = total;
	}
}
